src/index.js

```diff
@@ -31,6 +32,25 @@ const store = configureStore(reduxState)
 
 const history = syncHistoryWithStore(browserHistory, store)
 
+// token can be stored in localStorage in two scenario
+// 1. TWReporter account sign in
+// 2. oAuth
+// Acount: store auth info during signin action
+// oAuth: cookie -> redux state -> localStorage -> delete authinfo in redux state
+// The following procedure is only for oAuth
+const { auth } = store.getState()
+if(auth.authenticated && auth.authInfo && (auth.authType=== 'facebook' || auth.authType==='google')) {
+  setupTokenInLocalStorage(auth.authInfo)
+  store.dispatch(deletAuthInfoAction())
+}
+
+// Check if token existed in localStorage and expired
+// following preocedure is for both accoutn and oAuth SignIn
+// 7 = 7 days
+store.dispatch(authUserByTokenAction(7, auth.authType))
+  .then(() => {})
+  .catch(() => {})
+
```

Comment:
Do `authenticateUserByToken || authUserByTokenAction` need to be a `Promise`? 
Are [those actions](https://github.com/twreporter/react-redux-registration/blob/b3d885e7a545700240509e3ff3b13aa9bbf78190/src/actions/authentication.js#L137-L138) you send async functions?
If not async functions, you don't need to make it a Promise.

Reply:
👍